Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/MainActivity.java

```diff
@@ -4,149 +4,121 @@
 import android.support.v4.widget.DrawerLayout;
 import android.support.v7.app.ActionBarDrawerToggle;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.Toolbar;
 import android.view.MenuItem;
 import android.view.View;
-import android.widget.AdapterView;
-import android.widget.ListView;
 import com.mentalmachines.droidcon_boston.R;
 import com.mentalmachines.droidcon_boston.views.agenda.AgendaFragment;
 
 public class MainActivity extends AppCompatActivity {
 
   final FragmentManager fragmentManager = getSupportFragmentManager();
 
-  ActionBarDrawerToggle drawerToggle;
+  DrawerLayout androidDrawerLayout;
 
-  DrawerLayout mDrawerLayout;
+  ActionBarDrawerToggle actionBarDrawerToggle;
 
-  ListView mDrawerList;
+  NavigationView navigationView;
+
+  Toolbar toolbar;
 
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.main_activity);
 
-    Toolbar toolbar = findViewById(R.id.toolbar);
+    initNavDrawerToggle();
+
+    replaceFragment(new AgendaFragment(), getString(R.string.str_agenda));
+  }
+
+
+  private void initNavDrawerToggle() {
+
+    toolbar = findViewById(R.id.toolbar);
     setSupportActionBar(toolbar);
 
-    mDrawerLayout = findViewById(R.id.drawer_layout);
-    drawerToggle = new ActionBarDrawerToggle(
-        this, mDrawerLayout, toolbar, R.string.drawer_open, R.string.drawer_close);
-    mDrawerLayout.addDrawerListener(drawerToggle);
-    drawerToggle.syncState();
-    mDrawerList = findViewById(R.id.fragmentList);
-    mDrawerList.setAdapter(new NavigationAdapter(this));
-    mDrawerList.setOnItemClickListener(new DrawerItemClickListener());
-    fragmentManager.beginTransaction().replace(R.id.fragment_container, new AgendaFragment()).commit();
+    androidDrawerLayout = findViewById(R.id.drawer_layout);
+    actionBarDrawerToggle = new ActionBarDrawerToggle(MainActivity.this, androidDrawerLayout,
+        R.string.drawer_open, R.string.drawer_close);
+    androidDrawerLayout.addDrawerListener(actionBarDrawerToggle);
+
+    navigationView = findViewById(R.id.navView);
+    navigationView.setNavigationItemSelectedListener(item -> {
+
+      //Checking if the item is in checked state or not, if not make it in checked state
+      item.setChecked(!item.isChecked());
+
+      //Closing drawer on item click
+      androidDrawerLayout.closeDrawers();
+
+      switch (item.getItemId()) {
+        // Respond to the action bar's Up/Home button
+        case android.R.id.home:
+          if (fragmentManager.getBackStackEntryCount() > 0) {
+            fragmentManager.popBackStack();
+          } else if (fragmentManager.getBackStackEntryCount() == 1) {
+            // to avoid looping below on initScreen
+            super.onBackPressed();
+            finish();
+          }
+          break;
+        case R.id.nav_agenda:
+          replaceFragment(new AgendaFragment(), getString(R.string.str_agenda));
```

Comment:
should this look up the old frag from the fragment manager instead of always creating a new fragment?

Reply:
Fixed in recent commit pushed to this PR